ion-schema/src/main/kotlin/com/amazon/ionschema/internal/SchemaImpl_2_0.kt

```diff
@@ -27,46 +27,60 @@ import com.amazon.ionschema.IonSchemaVersion
 import com.amazon.ionschema.Schema
 import com.amazon.ionschema.Type
 import com.amazon.ionschema.Violations
-import com.amazon.ionschema.internal.util.ISL_2_0_RESERVED_WORDS_REGEX
+import com.amazon.ionschema.internal.util.IonSchema_2_0
+import com.amazon.ionschema.internal.util.getFields
 import com.amazon.ionschema.internal.util.islRequire
+import com.amazon.ionschema.internal.util.islRequireElementType
 import com.amazon.ionschema.internal.util.islRequireIonTypeNotNull
+import com.amazon.ionschema.internal.util.islRequireZeroOrOneElements
 import com.amazon.ionschema.internal.util.markReadOnly
 import kotlin.contracts.ExperimentalContracts
 import kotlin.contracts.contract
 
 /**
- * Implementation of [Schema] for all user-provided ISL.
- *
- * TODO: Handle user content in header, footer.
+ * Implementation of [Schema] for Ion Schema 2.0.
  */
 internal class SchemaImpl_2_0 private constructor(
     private val schemaSystem: IonSchemaSystemImpl,
     private val schemaCore: SchemaCore,
     schemaContent: Iterator<IonValue>,
     override val schemaId: String?,
     preloadedImports: Map<String, Import>,
+    preloadedUserReservedFields: UserReservedFields,
         /*
          * [types] is declared as a MutableMap in order to be populated DURING
          * INITIALIZATION ONLY.  This enables type B to find its already-loaded
          * dependency type A.  After initialization, [types] is expected to
          * be treated as immutable as required by the Schema interface.
          */
-    private val types: MutableMap<String, Type>
+    private val types: MutableMap<String, Type>,
 ) : SchemaImpl {
 
+    /**
+     * Represents reserved words that the user is reserving for their own open content.
```

Comment:
Is there a difference between a "reserved word" and a "reserved field?" If not, I suggest using a single term for consistency.

Also consider adding a comment with a link to [the relevant section of the spec](https://amzn.github.io/ion-schema/docs/isl-2-0/spec#open-content).

Reply:
There is a difference, and I'm using it incorrectly here. Good catch!

Reply:
```suggestion
     * Represents the collection of symbols that are declared as user reserved fields for a schema.
     * See https://amzn.github.io/ion-schema/docs/isl-2-0/spec#open-content
```